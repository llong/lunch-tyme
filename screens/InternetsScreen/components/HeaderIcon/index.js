// @flow

import React from 'react';
import { TouchableOpacity, Image } from 'react-native';

type Props = {
  icon: String
}

class HeaderIcon extends React.PureComponent<Props> {
  render() {
    const { icon, ...rest } = this.props;
    return (
      <TouchableOpacity {...rest}>
        <Image source={icon} style={{ marginRight: 16 }} />
      </TouchableOpacity>
    );
  }
}

export default HeaderIcon;
