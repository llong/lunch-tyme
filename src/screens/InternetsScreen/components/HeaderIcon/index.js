// @flow

import React from 'react';
import { TouchableOpacity, Image } from 'react-native';

type Props = {
  icon: any
}

class HeaderIcon extends React.PureComponent<Props> {
  render() {
    const { icon, ...rest } = this.props;
    return (
      <TouchableOpacity {...rest}>
        <Image source={icon} style={{ marginRight: 16, height: 24, width: 24, resizeMode: 'contain' }} />
      </TouchableOpacity>
    );
  }
}

export default HeaderIcon;
