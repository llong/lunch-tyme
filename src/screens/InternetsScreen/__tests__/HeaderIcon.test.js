import React from 'react';
import { shallow } from 'enzyme';
import HeaderIcon from '../components/HeaderIcon';


describe('HeaderIcon', () => {
  const wrapper = shallow(<HeaderIcon />);
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
