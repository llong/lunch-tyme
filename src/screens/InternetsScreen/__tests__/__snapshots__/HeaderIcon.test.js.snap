// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderIcon should render component 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <Image
    style={
      Object {
        "marginRight": 16,
      }
    }
  />
</TouchableOpacity>
`;
